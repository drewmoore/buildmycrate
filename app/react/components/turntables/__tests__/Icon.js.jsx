/* globals describe test expect */
import React          from 'react';
import renderer       from 'react-test-renderer';
import { mount }      from 'enzyme';
import TurntablesIcon from '../Icon.js.jsx';
import Images         from '../../../helpers/images.js.es6';
import '../../../__tests__/helpers/initialize.js.es6';

describe('TurntablesIcon', () => {
  const props = {
    side:          'left',
    loadTurntable: () => {}
  };
  const template = (<TurntablesIcon {...props} />);

  describe('DOM Snapshot', () => {
    const tree = renderer.create(template).toJSON();

    test('should match snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Rendering Props', () => {
    const icon = mount(template);

    test('should display left turntable image for prop side with value left', () => {
      const imageUrl = icon.find('[data-hook="turntable-icon"]').getDOMNode().src;
      expect(imageUrl).toEqual(Images.turntableLeft());
    });

    test('should display right turntable image for prop side with value right', () => {
      const newProps  = Object.assign({}, props, { side: 'right' });
      const otherIcon = mount(<TurntablesIcon {...newProps} />);
      const imageUrl  = otherIcon.find('[data-hook="turntable-icon"]').getDOMNode().src;
      expect(imageUrl).toEqual(Images.turntableRight());
    });
  });
});
