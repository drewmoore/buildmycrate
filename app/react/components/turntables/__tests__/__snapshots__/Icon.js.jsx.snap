// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TurntablesIcon DOM Snapshot should match snapshot 1`] = `
<img
  alt="Button to load left turntable"
  className="track-button clickable"
  data-hook="turntable-icon"
  onClick={[Function]}
  src="/turntable-left.jpg"
/>
`;
